enhancements/20231113-witness-node-support.md

```diff
@@ -0,0 +1,72 @@
+# Support witness node
+
+## Summary
+
+The witness node is a lightweight node that only runs the control plane components. The witness node is not schedulable and does not run any workload. The witness node forms a quorum with the other two nodes.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/3266
+
+## Motivation
+
```

Comment:
For a new reader, there are some open questions against this HEP:
(1) The motivation/background of such requirement; then it comes to this idea that such a `witness node` can be used to address certain issues
(2) When `witness node` mode is selected, could this node use less resource like <250g disk, <8 core CPU ? do we try to set a new resource limitation for this node?
(3) As Harvester is adding all container images from ISO to each NODE, both new installation and upgrade; then for `witness node`, it seems only system level images/specific images are required, do we set an image list for it? to reduce the resource consumption.
(4) Given a cluster with 2-control and 1-witness, what's the default replica-count of longhorn we suggest? the default `3` seems not fit for such scenario.
(5) Do we allow role change of witness node ? e.g. with fewer resource as witness to join, and then remove this role?
(6) Do we allow more than one witness node?
(7) The potential cross-affect of `rancher-vcluster`

Those points need to be clarified and defined clearly in the HEP.

Reply:
Thanks @w13915984028,  for the HEP suggestion.

I updated the latest HEP for the above questions. Please help to recheck it!